rest_framework_json_api/pagination.py

```diff
@@ -100,63 +116,34 @@ def get_paginated_response(self, data):
         })
 
 
-class JsonApiPageNumberPagination(JSONAPIPageNumberPagination):
-    """
-    Deprecated due to desire to use `JSONAPI` prefix for all classes.
-    """
-    page_query_param = 'page'
-    page_size_query_param = 'page_size'
-
-    def __init__(self):
-        warnings.warn(
-            'JsonApiPageNumberPagination is deprecated. Use JSONAPIPageNumberPagination '
-            'or create custom pagination. See '
-            'https://django-rest-framework-json-api.readthedocs.io/en/stable/usage.html#pagination',
-            DeprecationWarning)
-        super(JsonApiPageNumberPagination, self).__init__()
-
-
-class PageNumberPagination(JSONAPIPageNumberPagination):
+class PageNumberPagination(JsonApiPageNumberPagination):
     """
     Deprecated paginator that uses different query parameters
     """
     page_query_param = 'page'
     page_size_query_param = 'page_size'
 
     def __init__(self):
-        warnings.warn(
-            'PageNumberPagination is deprecated. Use JSONAPIPageNumberPagination '
-            'or create custom pagination. See '
-            'https://django-rest-framework-json-api.readthedocs.io/en/stable/usage.html#pagination',
-            DeprecationWarning)
+        if self.page_query_param == 'page' or self.page_size_query_param == 'page_size':
+            warnings.warn(
+                'PageNumberPagination will change in release 3.0 to default to use page[number] and'
```

Comment:
For deprecation warnings two things which I think are important:
1. Warning should be fixable at point of occurrence (so user should not wait till a new release to be able to fix a deprecation warning)
2. Warning should document how to fix them

To make it the case here we should explain that user should use custom pagination classes and they should define `page_query_param` and `page_size_query_param`.

For the deprecation warning it should only occur if the `type(self) == PageNumberPagination` or `type(self)`.__dict__ doesn't include `page_query_param` and `page_size_query_param`. (this codes bits are just out of my head and are not tested but I hope they explain what I mean.)

Reply:
Yes, thanks. I understand and will make it work.